python/lsst/ctrl/bps/report.py

```diff
@@ -127,17 +143,97 @@ def report(wms_service, run_id, user, hist_days, pass_thru, is_global=False, ret
 
             run_brief.clear()
             run_report.clear()
-        if not runs and not message:
+        if not runs and not messages:
             print(
                 f"No records found for job id '{run_id}'. "
-                f"Hints: Double check id, retry with a larger --hist value (currently: {hist_days}), "
+                f"Hints: Double check id, retry with a larger --hist value (currently: {hist}), "
                 "and/or use --global to search all job queues."
             )
     else:
+        runs, messages = retrieve_report(
+            wms_service,
+            user=user,
+            hist=hist,
+            pass_thru=pass_thru,
+            is_global=is_global,
+        )
         for run in runs:
             run_brief.add(run, use_global_id=is_global)
         run_brief.sort("ID")
         print(run_brief)
-    if message:
-        print(message)
+
+    if messages:
+        print("\n".join(messages))
         print("\n")
+
+
+def retrieve_report(
```

Comment:
This is the function that the campaign management tools will call?  I'm torn on passing postprocessors for the external API.  On one hand it provides nice flexibility.  On the other every time bps adds one for its report, campaign management tools will also have to modify their code.  Maybe a variable that has the list of postprocessors that bps generically uses in its call?  Then campaign management can use that until it needs to diverge.  I don't have a good mental picture of what all the postprocessors will be and whether bps report will use ones that other tools won't need and vice versa.

Reply:
Should we start a "main" API calls document?  i.e., mostly a page of pointers to the right function to which we can later add examples, hints, etc?

Reply:
I'll add a variable capturing the postprocessors that BPS uses.